api/v1alpha1/openapi.yaml

```diff
@@ -2529,6 +2632,22 @@ components:
               type: array
               items:
                 type: string
+        resources:
+          type: object
```

Comment:
Could we make this an array to eventually allow user-defined resource monitors (e.g. battery level)?

Reply:
so like oneOf

 type: array
  description: 'Array of resource monitor configurations.'
  items:
    type: object
    oneOf:
      - $ref: '#/components/schemas/ResourceCPUConfigSpec'
      - $ref: '#/components/schemas/ResourceMemoryConfigSpec'
      - $ref: '#/components/schemas/ResourceDiskConfigSpec'
    discriminator:
      propertyName: type
      mapping:
        cpu: '#/components/schemas/ResourceCPUConfigSpec'
        memory: '#/components/schemas/ResourceMemoryConfigSpec'
        disk: '#/components/schemas/ResourceDiskConfigSpec'

Reply:
will play with it

Reply:
I guess I don't understand how the user can update the api. You would need something like CRD right?

Reply:
Any monitor would require agent support so I'm not sure I understand how a generic API would be helpful the type would need to be extended based on support. Maybe you can provide a more concrete example.

Reply:
per conversation this would be future support